### ISSUE TITLE
Hotkeys set for source save replay are cleared on restart.

### ISSUE THREAD

**Issue by Tomshiii** [exeldro/obs-source-record#24]
Every time I boot obs, any custom keybinds I've set to save replays have been wiped and must be set again.
New to this whole thing, if you need anything from me lmk and I'll provide.

**Comment by exeldro** [comment]
Do you get an crash of OBS during shutdown that makes the hotkey not save?

**Comment by Tomshiii** [comment]
No, just a regular shutdown, either by pressing the exit button itself, or closing it on the top windows toolbar

**Comment by rse** [comment]
I can confirm this issue: When I set CTRL+ALT+I as the hotkey for the Source Record Save Replay, on exit I see in the `basic.ini` of my profile

```
[Hotkeys]
refresh_browsers.trigger={"bindings":[]}
...
ReplayBuffer={"ReplayBuffer.Save":[{"alt":true,"control":true,"key":"OBS_KEY_I"}]}
````
When I then restart OBS Studio and check the hotkey, it is gone again:

![image](https://user-images.githubusercontent.com/221273/125172107-ef995400-e1b7-11eb-9247-7ce28d880f85.png)

...although the entry in `basic.ini` actually is still there:

```
[Hotkeys]
refresh_browsers.trigger={"bindings":[]}
...
ReplayBuffer={"ReplayBuffer.Save":[{"alt":true,"control":true,"key":"OBS_KEY_I"}]}
```

So, it looks the hotkey actually IS persisted, but on restart not taken into account by Source Record.

**Comment by exeldro** [comment]
@rse the hotkey you found in basic.ini is the global replay buffer hotkey, not the hotkey from source_record. the hotkey should have the "save_replay" name instead of "ReplayBuffer.Save"

**Comment by rse** [comment]
Yes, correct, my fault. I tried to many thing in parallel today ;-) But there is no "save_replay", too.

**Comment by rse** [comment]
OK, I tried again from scratch: when I add the hotkey CTRL+ALT+I...

![image](https://user-images.githubusercontent.com/221273/125172383-c7125980-e1b9-11eb-86ce-f810b582e476.png)

...then it works as expected (the replay buffer is correctly saved) and even opening/closing the settings windows always shows that the hotkey is there. Once I exit OBS Studio I see NO entry in `basic.ini` and as a consequence the hotkey is gone the next time I open OBS Studio again. I also do NOT get any crash dumps, so either Source Record misses to save the hotkey in some way or OBS Studio crashes silently without performing a crash dump. Part of my game is just OBS Studio 27.0.1, Source Record 0.2.1, Source Switcher 0.3.0, Replay Source 1.6.9 and Source Dock 0.1.0.

**Comment by exeldro** [comment]
@rse it should be saved in the json file with the same name as the scene collection you are in like this `"save_replay":[{"alt":true,"control":true,"key":"OBS_KEY_I"}]`

**Comment by rse** [comment]
Oh, I see. Ok, but neverthelless, once I set the hotkey, close OBS Studio, I just see there:

```
..., "save_replay":[], ...
```

**Comment by rse** [comment]
And the issue is not related to the particular hotkey CTRL+ALT+I for me, as the hotkey is not persisted even when I use various other key combinations like CTRL+I or CTRL+K or whatever. I always just get this `"save_replay":[]` in my `.json` file.

**Comment by rse** [comment]
I don't know whether it helps, but in the same OBS Studio, as mentioned above, I also use your Replay Source 1.6.9 plugin and there two hotkeys. These(!) hotkeys are save just fine in the same `.json` file of my scene collection:

```
..., "ReplaySource.Replay":[{"alt":true,"control":true,"key":"OBS_KEY_L"}], ...
```

**Comment by spectangulation** [comment]
confirming i also lose my Source Record key binding on OBS restart. I am not using Record or Stream functions, only the replay buffer.

**Comment by spectangulation** [comment]
> confirming i also lose my Source Record key binding on OBS restart. I am not using Record or Stream functions, only the replay buffer.

actually sometimes, the Source Record hot key field is not present

**Comment by Tomshiii** [comment]
This is still an issue for me in later builds. If you require anything from me that could help you pin down this issue I would be more than happy to provide it. As mentioned in above comments, it doesn't seem to be saving in the json file post obs closure, no crashes.

**Comment by Lordmau5** [comment]
Gonna bump this once more because it happens to me as well.

Setting it to "Ctrl+Alt+Numpad9" and it gets cleared every time I launch OBS.

I'm on 0.2.6, as in the newest
